src/search/lp/lp_solver.cc

```diff
@@ -158,11 +214,36 @@ void LPSolver::load_problem(LPObjectiveSense sense,
           interfaces of all OSI versions <= 0.108.4 ignore it when it is
           set earlier. See issue752 for details.
         */
-        if (sense == LPObjectiveSense::MINIMIZE) {
+        if (lp.get_sense() == LPObjectiveSense::MINIMIZE) {
             lp_solver->setObjSense(1);
         } else {
             lp_solver->setObjSense(-1);
         }
+
+        if (!lp.get_objective_name().empty()) {
+            lp_solver->setObjName(lp.get_objective_name());
```

Comment:
Is it possible to set a name for an objective and not for variables and constraints? In this case `OsiNameDiscipline` will not be set below.